domain/learningandworkprogress/src/test/kotlin/uk/gov/justice/digital/hmpps/domain/learningandworkprogress/review/service/ReviewServiceTest.kt

```diff
@@ -79,4 +84,273 @@ class ReviewServiceTest {
+      val isReAdmission = false
+      val isTransfer = true
+
+      // When
+      val actual = service.determineReviewScheduleCalculationRule(releaseDate, sentenceType, isReAdmission, isTransfer)
+
+      // Then
+      assertThat(actual).isEqualTo(ReviewScheduleCalculationRule.PRISONER_TRANSFER)
+    }
+
+    @Test
+    fun `it should determine calculation rule given prisoner has an indeterminate sentence`() {
+      // Given
+      val releaseDate = null
+      val sentenceType = SentenceType.INDETERMINATE_SENTENCE
+      val isReAdmission = false
+      val isTransfer = false
+
+      // When
+      val actual = service.determineReviewScheduleCalculationRule(releaseDate, sentenceType, isReAdmission, isTransfer)
+
+      // Then
+      assertThat(actual).isEqualTo(ReviewScheduleCalculationRule.INDETERMINATE_SENTENCE)
+    }
+
+    @Test
+    fun `it should determine calculation rule given prisoner is on remand`() {
+      // Given
+      val releaseDate = null
+      val sentenceType = SentenceType.REMAND
+      val isReAdmission = false
+      val isTransfer = false
+
+      // When
+      val actual = service.determineReviewScheduleCalculationRule(releaseDate, sentenceType, isReAdmission, isTransfer)
+
+      // Then
+      assertThat(actual).isEqualTo(ReviewScheduleCalculationRule.PRISONER_ON_REMAND)
+    }
+
+    @Test
+    fun `it should determine calculation rule given prisoner is un-sentenced`() {
+      // Given
+      val releaseDate = null
+      val sentenceType = SentenceType.CONVICTED_UNSENTENCED
+      val isReAdmission = false
+      val isTransfer = false
+
+      // When
+      val actual = service.determineReviewScheduleCalculationRule(releaseDate, sentenceType, isReAdmission, isTransfer)
+
+      // Then
+      assertThat(actual).isEqualTo(ReviewScheduleCalculationRule.PRISONER_UN_SENTENCED)
+    }
+
+    @Test
+    fun `it should determine calculation rule given prisoner has less than 6 months left to serve`() {
+      // Given
+      val releaseDate = TODAY.plusMonths(6).minusDays(1)
+      val sentenceType = SentenceType.SENTENCED
+      val isReAdmission = false
+      val isTransfer = false
+
+      // When
+      val actual = service.determineReviewScheduleCalculationRule(releaseDate, sentenceType, isReAdmission, isTransfer)
+
+      // Then
+      assertThat(actual).isEqualTo(ReviewScheduleCalculationRule.LESS_THAN_6_MONTHS_TO_SERVE)
+    }
+
+    @Test
+    fun `it should determine calculation rule given prisoner has exactly 6 months left to serve`() {
+      // Given
+      val releaseDate = TODAY.plusMonths(6)
+      val sentenceType = SentenceType.SENTENCED
+      val isReAdmission = false
+      val isTransfer = false
+
+      // When
+      val actual = service.determineReviewScheduleCalculationRule(releaseDate, sentenceType, isReAdmission, isTransfer)
+
+      // Then
+      assertThat(actual).isEqualTo(ReviewScheduleCalculationRule.LESS_THAN_6_MONTHS_TO_SERVE)
```

Comment:
This is the only one I am not 100% happy with (but it's only semantics)
The business rules use the wording "6 months or less", but I'd created the enum in a previous PR, and I'd named this particular enum value `LESS_THAN_6_MONTHS_TO_SERVE`, which in this specific case is wrong! (all the other enum values are named correctly I think, it's just this one)

It's just a naming refactor, but it'll affect half a dozen files, so I'll probably rename this enum value in a separate PR